pkg/provisioning/keypair.go

```diff
@@ -0,0 +1,133 @@
+package provisioning
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	privateKeyFile = "serviceaccount-signer.private"
+	publicKeyFile  = "serviceaccount-signer.public"
+)
+
+func createKeys(prefixDir string) {
+
+	privateKeyFilePath := filepath.Join(prefixDir, privateKeyFile)
+	publicKeyFilePath := filepath.Join(prefixDir, publicKeyFile)
+	bitSize := 4096
+
+	_, err := os.Stat(privateKeyFilePath)
+	if err == nil {
+		log.Printf("Using existing RSA keypair found at %s", privateKeyFilePath)
+		return
+	}
```

Comment:
Is it okay to assume that if private key exists, public will also exist?

Reply:
I thought about this a bit. And I'm not sure what the right answer is. We can always regenerate the public key from the private key, so maybe we don't even need to save the public key file at this point. Or the future code that needs the public key can always just generate it from the private key as needed. :shrug: